src/System.cpp

```diff
@@ -141,6 +146,23 @@ namespace System
         return temp;
     }
 
+    double GetBatteryPercentage()
+    {
+        std::ifstream fullChargeFile(config.batteryFolder + "/charge_full");
+        std::ifstream currentChargeFile(config.batteryFolder + "/charge_now");
+        if (!fullChargeFile.is_open() || !currentChargeFile.is_open())
+        {
+            return 0.f;
```

Comment:
Can you make the result negative on error? Since *technically* (although very unlikely) 0 is a valid percentage.